ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -2239,47 +2239,62 @@ select PartCentredForceHolder.
 
   private void PlotCelestialTrajectories(DisposablePlanetarium planetarium,
                                          string main_vessel_guid) {
-    foreach (CelestialBody celestial in FlightGlobals.Bodies) {
-      if (plotting_frame_selector_.FixesBody(celestial)) {
-        continue;
-      }
-      var colour = celestial.orbitDriver?.Renderer?.orbitColor ??
-                   XKCDColors.SunshineYellow;
-      if (colour.a != 1) {
-        // When zoomed into a planetary system, the trajectory of the
-        // planet is hidden in stock (because KSP then draws most things
-        // in the reference frame centred on that planet).
-        // Here we still want to display the trajectory of the primary,
-        // e.g., if we are drawing the trajectories of the Jovian system
-        // in the heliocentric frame.
-        foreach (CelestialBody child in celestial.orbitingBodies) {
-          colour.a = Math.Max(
-              child.MapObject?.uiNode?.VisualIconData.color.a ?? 1,
-              colour.a);
-        }
-      }
-      if (colour.a == 0) {
-        continue;
-      }
+    // TODO(egg): use that angular resolution in the planetarium too.
+    const double degree = Math.PI / 180;
+    UnityEngine.Camera camera = PlanetariumCamera.Camera;
+    // The angle subtended by the pixel closest to the centre of the viewport.
+    double tan_angular_resolution =
+        Math.Tan(camera.fieldOfView * degree / 2) / (camera.pixelHeight / 2);
```

Comment:
Pixels may not be square, I'd take the min of the values obtained from the pixel height and width.

Reply:
No. `camera.fieldOfView` is the vertical field of view.